include/gatekeeper_ggu.h

```diff
@@ -19,11 +19,101 @@
 #ifndef _GATEKEEPER_GGU_H_
 #define _GATEKEEPER_GGU_H_
 
+#include "gatekeeper_net.h"
+#include "gatekeeper_flow.h"
+
 /* Configuration for the GK-GT Unit functional block. */
 struct ggu_config {
-	unsigned int	lcore_id;
+	unsigned int      lcore_id;
+
+	/* The UDP source and destination port numbers for GGU. */
+	uint16_t          ggu_src_port;
+	uint16_t          ggu_dst_port;
+
+	/*
+	 * The fields below are for internal use.
+	 * Configuration files should not refer to them.
+	 */
+
+	/* RX queue on the back interface. */
+	uint16_t          rx_queue_back;
+	struct net_config *net;
+	struct gk_config  *gk;
+};
+
+/*
+ * Since the length of IPv6 policy is much larger than IPv4 policy,
+ * to save network bandwidth, we choose to use different data structures
+ * to store the in-packet policies for IPv4 and IPv6, respectively.
+ *
+ * Packet format: Ethernet headers + IP header + UDP header + Data.
+ * In the UDP payload, the following format would save a lot of bytes
+ * when there are decline decisions:
+ *  v1, n1, n2, n3, n4: Each of these fields are 1-byte long.
+ *  v1 is a constant indicating the version of the format, in this case 1.
+ *  n1 is the number of IPv4 decline decisions.
+ *  n2 is the number of IPv6 decline decisions.
+ *  n3 is the number of IPv4 granted decisions.
+ *  n4 is the number of IPv6 granted decisions.
+ * 
+ * Field v1 will enable us to change the format, incrementally update
+ * the Gatekeeper servers, and incrementally update the Grantor servers.
+ */
+struct ggu_common_hdr {
+	uint8_t v1;
+	uint8_t n1;
+	uint8_t n2;
+	uint8_t n3;
+	uint8_t n4;
+}__attribute__((packed));
```

Comment:
Please add `uint8_t reserved[3];` at the very end of `struct ggu_common_hdr`. This will guarantee that all the accesses after `struct ggu_common_hdr` in a packet are 32-bit aligned.

Don't forget to update the comment about the format above.